### PR TITLE
refactor: addressing small audit findings

### PR DESCRIPTION
**Pull request by kupermind**
- Addressing small Cantina audit findings.

### CODE REVIEW

File: lockbox/programs/liquidity_lockbox/src/lib.rs [hunk 1]
```diff
@@ -131,7 +126,7 @@ pub mod liquidity_lockbox {
     }
 
     // Check the lockbox PDA address correctness
-    let lockbox_pda = Pubkey::find_program_address(&[b"liquidity_lockbox"], &PROGRAM_ID);
+    let lockbox_pda = Pubkey::find_program_address(&[b"liquidity_lockbox"], &ID);
```
Comment (kupermind) [inline lockbox/programs/liquidity_lockbox/src/lib.rs:129]: The program ID is already accessible by just the `ID` variable.

File: lockbox/programs/liquidity_lockbox/src/lib.rs [hunk 2]
```diff
@@ -401,22 +396,7 @@ pub mod liquidity_lockbox {
...
-
-      let mut data = account.try_borrow_mut_data()?;
-      for byte in data.deref_mut().iter_mut() {
-        *byte = 0;
-      }
-
-      let dst: &mut [u8] = &mut data;
-      let mut cursor = Cursor::new(dst);
-      cursor.write_all(&CLOSED_ACCOUNT_DISCRIMINATOR).unwrap();
+      ctx.accounts.pda_lockbox_position.close(ctx.accounts.signer.to_account_info())?;
```
Comment (kupermind) [inline lockbox/programs/liquidity_lockbox/src/lib.rs:399]: Change manual account closing to a new `close()` function provided out of the box.

File: lockbox/programs/liquidity_lockbox/src/lib.rs [hunk 3]
```diff
@@ -503,11 +483,11 @@ pub struct DepositPositionForLiquidity<'info> {
     payer = signer)]
   pub pda_lockbox_position: Box<Account<'info, LockboxPosition>>,
 
-  #[account(mut)]
+  #[account(mut, address = lockbox.bridged_token_mint)]
   pub bridged_token_mint: Box<Account<'info, Mint>>,
   #[account(mut,
     constraint = bridged_token_account.mint == lockbox.bridged_token_mint,
-    constraint = bridged_token_mint.key() == lockbox.bridged_token_mint,
+    constraint = bridged_token_account.mint == bridged_token_mint.key(),
```
Comment (kupermind) [inline lockbox/programs/liquidity_lockbox/src/lib.rs:490]: Switching context of verified accounts to correspond to actual declared accounts.

File: lockbox2/programs/liquidity_lockbox/src/lib.rs [hunk 4]
```diff
@@ -57,7 +55,9 @@ pub mod liquidity_lockbox {
     let account = &ctx.accounts.position.to_account_info();
 
     let data = account.try_borrow_data()?;
-    assert!(data.len() > 8);
+    if data.len() != Position::LEN {
+        return Err(ErrorCode::WrongPositionHeader.into());
+    }
```
Comment (kupermind) [inline lockbox2/programs/liquidity_lockbox/src/lib.rs:60]: Verifying the position account length.

File: lockbox2/programs/liquidity_lockbox/src/lib.rs [hunk 5]
```diff
@@ -168,12 +168,6 @@ pub mod liquidity_lockbox {
       return Err(ErrorCode::WrongTokenMint.into());
     }
 
-    // Check tick arrays owner
-    if ctx.accounts.tick_array_lower.to_account_info().owner != &ORCA ||
-      ctx.accounts.tick_array_upper.to_account_info().owner != &ORCA {
-      return Err(ErrorCode::WrongOwner.into());
-    }
```
Comment (kupermind) [inline lockbox2/programs/liquidity_lockbox/src/lib.rs:175]: Obsolete as done on a context verification level.

File: lockbox2/programs/liquidity_lockbox/src/lib.rs [hunk 6]
```diff
@@ -185,10 +179,12 @@ pub mod liquidity_lockbox {
...
     // Total liquidity update with the check
-    ctx.accounts.lockbox.total_liquidity = ctx.accounts.lockbox
+    ctx.accounts.lockbox.total_liquidity = match ctx.accounts.lockbox
       .total_liquidity
-      .checked_add(liquidity_amount)
-      .unwrap_or_else(|| panic!("Liquidity overflow"));
+      .checked_add(liquidity_amount) {
+        Some(new_liquidity) => new_liquidity,
+        None => return Err(ErrorCode::LiquidityOverflow.into()),
+      };
```
Comment (kupermind) [inline lockbox2/programs/liquidity_lockbox/src/lib.rs:187]: Beautifying the `checked_add()` call failure.

File: lockbox2/programs/liquidity_lockbox/src/lib.rs [hunk 7]
```diff
@@ -320,7 +316,7 @@ pub mod liquidity_lockbox {
     }
 
     // Check the lockbox PDA address correctness
-    let lockbox_pda = Pubkey::find_program_address(&[b"liquidity_lockbox"], &PROGRAM_ID);
+    let lockbox_pda = Pubkey::find_program_address(&[b"liquidity_lockbox"], &ID);
```
Comment (kupermind) [inline lockbox2/programs/liquidity_lockbox/src/lib.rs:319]: Program ID is already written in just the `ID` variable.

File: lockbox2/programs/liquidity_lockbox/src/lib.rs [hunk 8]
```diff
@@ -340,12 +336,6 @@ pub mod liquidity_lockbox {
       return Err(ErrorCode::WrongTokenMint.into());
     }
 
-    // Check tick arrays owner
-    if ctx.accounts.tick_array_lower.to_account_info().owner != &ORCA ||
-      ctx.accounts.tick_array_upper.to_account_info().owner != &ORCA {
-      return Err(ErrorCode::WrongOwner.into());
-    }
```
Comment (kupermind) [inline lockbox2/programs/liquidity_lockbox/src/lib.rs:347]: Obsolete as checks are provided on a context verification level.

File: lockbox2/programs/liquidity_lockbox/src/lib.rs [hunk 9]
```diff
@@ -504,6 +496,7 @@ pub struct DepositPositionForLiquidity<'info> {
   pub position: Box<Account<'info, Position>>,
 
   #[account(mut,
+    address = lockbox.pda_position_account.key(),
```
Comment (kupermind) [inline lockbox2/programs/liquidity_lockbox/src/lib.rs:499]: Added a check for the `pda_position_account` to be the one that is in the `lockbox` account.

File: lockbox2/programs/liquidity_lockbox/src/lib.rs [hunk 10]
```diff
@@ -533,16 +526,21 @@ pub struct DepositPositionForLiquidity<'info> {
...
-  #[account(mut, has_one = whirlpool, constraint = tick_array_lower.key() != tick_array_upper.key())]
+  #[account(mut, has_one = whirlpool,
+    constraint = tick_array_lower.key() != tick_array_upper.key(),
+    constraint = tick_array_lower.to_account_info().owner == &whirlpool_program.key()
+  )]
   pub tick_array_lower: AccountLoader<'info, TickArray>,
-  #[account(mut, has_one = whirlpool)]
+  #[account(mut, has_one = whirlpool,
+    constraint = tick_array_upper.to_account_info().owner == &whirlpool_program.key()
+  )]
```
Comment (kupermind) [inline lockbox2/programs/liquidity_lockbox/src/lib.rs:536]: Tick checks.

File: lockbox2/programs/liquidity_lockbox/src/lib.rs [hunk 11]
```diff
@@ -533,16 +526,21 @@ pub struct DepositPositionForLiquidity<'info> {
...
   pub tick_array_upper: AccountLoader<'info, TickArray>,
 
   #[account(mut, address = lockbox.bridged_token_mint)]
   pub bridged_token_mint: Box<Account<'info, Mint>>,
   #[account(mut,
     constraint = bridged_token_account.mint == lockbox.bridged_token_mint,
-    constraint = bridged_token_mint.key() == lockbox.bridged_token_mint,
+    constraint = bridged_token_account.mint == bridged_token_mint.key(),
     constraint = signer.key == &bridged_token_account.owner,
```
Comment (kupermind) [inline lockbox2/programs/liquidity_lockbox/src/lib.rs:544]: Switching context of verified accounts to correspond to actual declared accounts.

File: lockbox2/programs/liquidity_lockbox/src/lib.rs [hunk 12]
```diff
@@ -552,9 +550,7 @@ pub struct DepositPositionForLiquidity<'info> {
   pub whirlpool_program: Program<'info, whirlpool::program::Whirlpool>,
 
   #[account(address = token::ID)]
-  pub token_program: Program<'info, Token>,
-  pub system_program: Program<'info, System>,
-  pub rent: Sysvar<'info, Rent>,
```
Comment (kupermind) [inline lockbox2/programs/liquidity_lockbox/src/lib.rs:557]: Not needed.

File: lockbox2/programs/liquidity_lockbox/src/lib.rs [hunk 13]
```diff
@@ -568,14 +564,15 @@ pub struct WithdrawLiquidityForTokens<'info> {
   pub bridged_token_mint: Box<Account<'info, Mint>>,
   #[account(mut,
     constraint = bridged_token_account.mint == lockbox.bridged_token_mint,
-    constraint = lockbox.bridged_token_mint == bridged_token_mint.key(),
+    constraint = bridged_token_account.mint == bridged_token_mint.key(),
     constraint = signer.key == &bridged_token_account.owner,
   )]
   pub bridged_token_account: Box<Account<'info, TokenAccount>>,
```
Comment (kupermind) [inline lockbox2/programs/liquidity_lockbox/src/lib.rs:570]: Switching context of verified accounts to correspond to actual declared accounts.

**Review by 77ph — APPROVED** [review]

**Review by mariapiamo — APPROVED** [review]

**Pull request merged**